bedita-app/vendors/shells/tasks/cleanup_data.php

```diff
@@ -0,0 +1,134 @@
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * BEdita is distributed WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with BEdita (see LICENSE.LGPL).
+ * If not, see <http://gnu.org/licenses/lgpl-3.0.html>.
+ *
+ *------------------------------------------------------------------->8-----
+ */
+
+/**
+ * Cleanup data task
+ * 
+ * @property \BEobject $BEobject
+ */
+class CleanupDataTask extends BeditaBaseShell {
+    
+    /**
+     * @inheritDoc
+     */
+    public $uses = ['BEObject'];
+
+    protected $truncateTables = ['event_logs', 'mail_jobs', 'mail_logs'];
+
+    protected $cleanupTables = ['history', 'versions'];
+
+    public function help() {
+        $this->hr();
+        $this->out('cleanup data script shell usage:');
+        $this->out('');
+        $this->out('./cake.sh cleanupData // truncate event_logs, mail_jobs, mail_logs, clean old data from history and versions');
+        $this->out('./cake.sh cleanupData -ld <limit date yyyy-MM-dd> // use custom limit date for clean');
+        $this->out('./cake.sh cleanupData -ni // no interactive mode');
+        $this->out('./cake.sh cleanupData -tt <comma separated table names> // truncate event_logs, mail_jobs, mail_logs + more tables');
+        $this->out('./cake.sh cleanupData help // show this help');
+        $this->out('');
+    }
+
+    /**
+     * Tables to clean (truncate or delete data):
+     * - event_logs (truncate)
+     * - mail_jobs (truncate, if no messages pending)
+     * - mail_logs (truncate)
+     * - history (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     * - versions (cleanup: mantain only "CY-1 / CY" data (CY is the Current Year))
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        $this->hr();
+        $this->out('BEdita Data Cleanup');
+        $this->out('--- count data before cleaning ---');
+        $tables = $this->truncateTables;
+        if (isset($this->params['tt'])) {
+            $tables = array_merge($tables, explode(',', $this->params['tt']));
+        }
+        $countTables = array_merge($this->cleanupTables, $tables);
+        sort($countTables);
+        $limitDate = date('Y')-2 . '-12-31 23:59:59';
+		if (isset($this->params['ld'])) {
+			$limitDate = $this->params['ld'] . ' 23:59:59';
+        }
+        $limitWhere = sprintf('WHERE created < \'%s\'', $limitDate);
+        foreach ($countTables as $tableName) {
+            $this->countRecords($tableName);
+            $this->countRecords($tableName, $limitWhere);
+        }
```

Comment:
I think it's better to use `$limitWhere` only for those tables where it's actually used, either by using 2 separate loops for `$tables` and `$this->cleanupTables` or by adding conditions to handle with a single loop, i.e.:

```suggestion
        foreach ($countTables as $tableName) {
            if (in_array($tableName, $this->cleanupTables) {
                $this->countRecords($tableName, $limitWhere);
            } else {
                $this->countRecords($tableName);
            }
        }
```

Reply:
well, the idea was to count all data, then cleanup using the limit. I could make it clearer with 2 counts (without and with limit)